### DISCUSSION TITLE
Annotating and saving a molecule

### DISCUSSION THREAD

**Discussion by EqualAPriori** [rdkit/rdkit#6181] [Q&A]
Hi all, 

Thank you for developing this great tool! I'm looking to annotate molecules -- not just visually in an image which I see tutorials of, but to save those annotations in an reloadable way as well. The idea is I want to be able to "save notes" about molecules (and groups of atoms) that I'm looking at.

I was wondering if rdkit has an easy way to do this -- I'm new to rdkit, and from reading documentation and searching the below is what I've pieced together so far:

1. I'll need a portable and persistent atom ordering --> I may generate the initial structure via a SMILES, but I should probably save as a .mol, .sdf file, or canonical smiles
2. I may have missed this, but I couldn't find how rdkit can save molecules (beyond pickling), so it's probably easiest to save an additional annotation file *alongside* the molecule representation from (1) above, e.g. a list of ( (atom indices), "annotation" ) tuples.

Alternative:
1. use CXSMILES. under atom properties, hold a list of "note reference id"
2. in a separate variable/file, hold a list of (note reference id, "annotation") tuples.

This might be a bit more portable, as the atom indices aren't explicitly saved. However, while selecting atoms I'd probably still have to use atom index numbers, probably generated on the fly.

Lastly, might someone have already put together a simple GUI wrapping RDKit to make atom selection easier? E.g. in Ketcher I know you can select and highlight atoms, but I don't know of any way of getting the atom numbers out. Instead, the best I can figure out right now is to display the atom numbers via rdkit, and then "manually retrieving" the atom numbers that I desire.

Thanks!

**Comment by JanoschMenke** [comment]
You can always write additional molecular Information to sdf files.

```python
from rdkit.Chem import AllChem as Chem
supplier = Chem.SDMolSupplier('data/molecules.sdf')
sdfWriter = Chem.SDWriter('data/newSDF.sdf') 
for m in supplier:
    m.SetProp('example', .2342)     
    sdfWriter.write(m)   
sdfWriter.close()
```

For atom specific information you would maybe need to be use tuples
